### PR TITLE
[Feature] Add Name/Data Node Health Checks

### PR DESCRIPTION
**Pull request by SayedAlesawy**

### CODE REVIEW

File: config/data_node.go [hunk 1]
```diff
@@ -10,6 +10,8 @@ type DataNodeconfig struct {
 	NameNodeIP                   string //IP of the current name node
 	InternalRequestsPort         string //The internal requests ports
 	NameNodeInternalRequestsPort string //The internal requests port of the name node
+	NetowrkProtocol              string //Network protocol used by the data node
```
Comment (ahmibr) [inline config/data_node.go]: typo

File: name_node/health.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"time"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+	"github.com/SayedAlesawy/Videra-Storage/data_node/dnpb"
+	"google.golang.org/grpc"
+)
+
+// PingDataNodes A function to ping all currently conneced data nodes for health checking
+func (nameNode *NameNode) PingDataNodes() {
+	time.Sleep(10 * time.Second)
```
Comment (ahmibr) [inline name_node/health.go]: Maybe this will be moved later to config files as readiness time

File: name_node/health.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	for range time.Tick(nameNode.HealthCheckInterval) {
+		for _, dataNode := range nameNode.DataNodes {
+			address := nameNode.getNameNodeAddress(dataNode)
+
+			conn, err := grpc.Dial(address, grpc.WithInsecure())
+			defer conn.Close()
+			if errors.IsError(err) {
```
Comment (ahmibr) [inline name_node/health.go:22]: I think this may need handling same as when sending health check fails

File: config/name_node.go [hunk 4]
```diff
@@ -9,6 +9,8 @@ type NameNodeconfig struct {
 	IP                   string //Name node IP
 	InternalRequestsPort string //The internal requests ports
 	NetowrkProtocol      string //Name network protcol
```
Comment (ahmibr) [inline config/name_node.go]: typo

File: name_node/health.go (hunk 2 above)
Comment (SayedAlesawy) [reply name_node/health.go]: This is useless, I was testing something. Removed it.

File: name_node/health.go (hunk 3 above)
Comment (SayedAlesawy) [reply name_node/health.go:22]: What type of handling are you suggesting? This basic comm btw, this PR doesn't include the actions that will be taken if a node is down, like removing it from the list of alive nodes and so on.

Comment (ahmibr) [reply name_node/health.go:22]: I'm just noting that logic will be inserted here also when implemented

**Review by ahmibr — APPROVED** [review]

**Pull request merged**